bin/index.js

```diff
@@ -3,61 +3,83 @@
 const shell = require('shelljs')
 const request = require('superagent')
 
-const alias = process.argv[2].toLowerCase()
+const alias = process.argv[2]?.toLowerCase()
 const listingUrl = process.argv[3]
 
 if (!alias) {
   showUsage()
 } else if (alias === '--init') {
-  setupExtension(listingUrl)
+  const isOnCampus = process.argv[4] === '--on-campus'
+  setupExtension(listingUrl, isOnCampus)
 } else {
   configureUser(alias)
 }
 
-function showUsage () {
+function showUsage() {
   console.info(`Usage:
-  git-iam --init url
-    Sets up the extension. The url should be the address of a JSON file.
+  git-iam --init url [--on-campus]
+    Sets up the extension. The url should be the address of a JSON file. Pass the --on-campus flag if you are on campus.
   git iam user
     Sets the name and email config of the user based on the JSON file.`)
 }
 
-function setupExtension (url) {
+function setupExtension(url, isOnCampus) {
```

Comment:
I'm thinking of adding a default parameter like this `function setupExtension(url, isOnCampus = true)`, so that when the flag is not explicitly passed from the CLI it automatically assumes that it's a campus environment (to opt-in).

Reply:
I'd rather see the default as isOnCampus = false as overwriting an online students global config strikes me as a worse outcome than failing to overwrite the config on a shared computer.

Reply:
My rationale was that a user should have to opt-in to the more invasive behaviour (modifying potentially personal files like a global gitconfig)

Reply:
@rohan-fowlerharper does this mean that we (akl and welli) don't need to use a flag when configuring it?
I would prefer to have it like this:
```
# online
git-iam --init https://gist.githubusercontent.com/example-user/123example-id/raw/ --online
```
```
# akl and welli
git-iam --init https://gist.githubusercontent.com/example-user/123example-id/raw/
```

Reply:
Like JV mentioned, `isOnCampus === true` removes the student's personal git config, hence the choice to have to opt-in to that behaviour rather than the other way round.

A concern of mine would be a student copying the link themselves and leaving off an `--online` flag and then losing their git config unknowingly.

I wouldn't mind a compromise where a flag of `--personal`/`--shared` or `--online`/`--on-campus` is required to init it. Thoughts?

Reply:
My sense is that online teachers would like online to be the default and in person teachers would like in person to be the default. I think roh's argument that the least destructive option should be the default is sound - I don't mind what the flag is called.

Reply:
Now I understand.

Reply:
Not running init without an explicit flag passed is mostly fine, but if someone updates docs in some manner, even with examples, that students could copy/paste, it could accidentally result in the destructive behaviour.

Alternatively, have a bold `**WARNING:**` or similar note next to the example for a student to see and be aware of if defaulting (or the example if copied would) to a destructive behaviour.

I personally avoided installing the package because of this concern on my personal machine (online cohort) where I have a git profile already for volunteer work prior to joining EDA. I would imagine that this is far less likely a concern on campus, and for that you're better off just providing an ENV to those machines that provides the preferred campus flag by default.